builtin/game/knockback.lua

```diff
@@ -0,0 +1,45 @@
+-- can be overriden by mods
+core.calculate_knockback = function(player, hitter, time_from_last_punch, tool_capabilities, dir, distance, damage)
+	local m = 8
+	-- solve m - m*e^(k*4) = 4 for k
+	local k = -0.17328
+	local res = m - m * math.exp(k * damage)
+
+	if distance < 2.0 then
+		res = res * 1.1 -- more knockback when closer
+	elseif distance > 4.0 then
+		res = res * 0.9 -- less when far away
+	end
+	return res
+end
+
+local function vector_absmax(v)
+	local max, abs = math.max, math.abs
+	return max(max(abs(v.x), abs(v.y)), abs(v.z))
+end
+
+core.register_on_punchplayer(function(player, hitter, time_from_last_punch, tool_capabilities, unused_dir, damage)
+	if player:get_hp() == 0 then
+		return -- RIP
```

Comment:
Kicking around dead corpses might be fun or even useful. Do we really want to return here?

Reply:
All movement is ignored when a player is dead, so this wouldn't work anyway.